### ISSUE TITLE
PhpDoc return types

### ISSUE THREAD

**Issue by tdtm** [spatie/icalendar-generator#32]
Hello,

Analysing my codebase with Psalm after using ths package, I ran into issues where I was warned of UndefinedMethods.

I raised that with the `psalm` team here: https://github.com/vimeo/psalm/issues/4263

Something like this causes issues:
```php
        $event = Event::create()
            ->uniqueIdentifier($uid)
            ->startsAt($start)
            ->endsAt($end)
            ->name($summary)
            ->description($description)
            ->appendProperty(TextPropertyType::create('URL', $activity_url))
            ->attendee($member['email'], null, $participationStatus);

        if ($address) {
            $event->address(address);
        }

        if ($lat && $lng) {
            $event->coordinates($lat, $lng);
        }

		$calendar->appendSubComponent($event);
```

Essentially all because `appendProperty()` is typed to return `Component`, and so the calls that follow (`attendee()`, `address()`, `coordinates()`) all fail as those are methods of `Event`, not of `Component`.

There are two solutions to the issue:
- move `appendProperty()` to be the last statement before `$calendar->appendSubComponent($event)`
- add `static` as a return type to most fluent calls.

PHP8 will introduce `static` as an actual return type - which solves the issue - but in lower versions, the only solution is really to provide this in phpdoc alongside the `Component` use in the method signature.

So as an example:

```php
    public function appendProperty(PropertyType $property): Component
    {
        $this->appendedProperties[] = $property;

        return $this;
    }
```

Would simply become:
```php
	/**
	 * @return static
	 */
    public function appendProperty(PropertyType $property): Component
    {
        $this->appendedProperties[] = $property;

        return $this;
    }
```

**bottom line: if submit a PR for that, would this be considered?**

From my tests, this applies to Psalm and Phpstan. I haven't tested how Phan and others consider it. For PhpStorm's built-in analysis, it doesn't need the `@return static` return type, but it doesn't break anything either.

**Comment by tdtm** [comment]
Ref: my issue discussing that in the psalm repo: https://github.com/vimeo/psalm/issues/4263

**Comment by rubenvanassche** [comment]
I'm all in for better types! I've just added a psalm GitHub Action and added Psalm to the project. So that PR would certainly be welcome!
